gradio/routes.py

```diff
@@ -448,12 +458,42 @@ async def predict(
                 gr_request = Request(request)
             result = await run_predict(
                 body=body,
-                fn_index_inferred=fn_index_inferred,
+                fn_index=fn_index,
                 username=username,
                 request=gr_request,
             )
             return result
 
+        @app.post("/run/binary/{api_name}", dependencies=[Depends(login_check)])
+        async def predict_binary(
+            api_name: str,
+            request: fastapi.Request,
+            payload: str = Form(...),
+            binary_files: List[UploadFile] = File(...),
```

Comment:
Something to discuss at the synch: Did you try adding `binary_files` argument to the standard `run/predict/` route? Think would be simpler for api users and us as maintainers if there was just one route.